### ISSUE TITLE
Provide the ability to set redirect when calling `Client.Instance.Auth.SendMagicLink` or `Client.Instance.Auth.SignIn`

### ISSUE THREAD

**Issue by MisterJimson** [supabase-community/supabase-csharp#20]
# Feature request
Provide the ability to set redirect when calling `Client.Instance.Auth.SendMagicLink` or `Client.Instance.Auth.SignIn`

Other Supabase client libs include this and it would help for deep linking magic link emails to the application.

**Comment by acupofjose** [comment]
The SignUp method currently accepts [SignInOptions.RedirectTo](https://supabase-community.github.io/supabase-csharp/api/Supabase.Gotrue.SignUpOptions.html) but you’re right, we don’t have that for `SendMagicLink` And it should be added!

**Comment by MisterJimson** [comment]
May want to transfer the issue to `gotrue-csharp`, it doesn't allow me to.

**Comment by Kieudung717** [comment]
> # Feature request
> Provide the ability to set redirect when calling `Client.Instance.Auth.SendMagicLink` or `Client.Instance.Auth.SignIn`
> 
> Other Supabase client libs include this and it would help for deep linking magic link emails to the application.

**Comment by acupofjose** [comment]
This is available in `>= supabase-csharp@0.3.1` `supabase-community/gotrue-csharp@2.3.5`

```c#
var options = new Supabase.Gotrue.SignInOptions { 
  RedirectTo = "https://my-super-url.com" 
};
await Supabase.Client.Instance.Auth.SendMagicLink("michael@supabase.io", options);
```
And

```c#
var options = new Supabase.Gotrue.SignInOptions { 
  RedirectTo = "https://my-super-url.com" 
};
await Supabase.Client.Instance.Auth.SignIn("michael@supabase.io", options);
```
